packages/algob/src/internal/deployer.ts

```diff
@@ -603,19 +648,41 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
 
   /**
    * This function will send Algos to ASC account in "Contract Mode"
-   * @param name     - ASC filename
+   * @param fileName     - ASC filename
    * @param flags    - Deployments flags (as per SPEC)
    * @param payFlags - as per SPEC
    * @param scTmplParams: Smart contract template parameters (used only when compiling PyTEAL to TEAL)
    */
   async fundLsig (
-    name: string,
+    fileName: string,
     flags: FundASCFlags,
     payFlags: wtypes.TxParams,
     scTmplParams?: SCParams
   ): Promise<void> {
     try {
-      await this.algoOp.fundLsig(name, flags, payFlags, this.txWriter, scTmplParams);
+      await this.algoOp.fundLsig(fileName, flags, payFlags, this.txWriter, scTmplParams);
+    } catch (error) {
+      console.log(error);
+      throw error;
+    }
+  }
+
+  /**
+   * This function will send Algos to ASC account in "Contract Mode". Takes lsig name
+   * as input
+   * @param lsigName - name of the lsig (passed by user during mkContractLsig/mkDelegatedLsig)
+   * @param flags    - Deployments flags (as per SPEC)
+   * @param payFlags - as per SPEC
+   * @param scTmplParams: Smart contract template parameters (used only when compiling PyTEAL to TEAL)
+   */
+  async fundLsigByName (
```

Comment:
If we really default is teal file name. Why we need both fundLsig and fundLsigByName?. Are they can be one? 
Should we create `fundToAddress`  use for application, lsig and account address?

Reply:
The idea is to promote using app/lsig names, we can't simply remove `fundLsig` because we already have it. Also passing/using names is optional. So `fundLsig` would work in that case.

Reply:
How is a `fundToAddress` different from an AlgoTransfer transaction ?

Reply:
> How is a `fundToAddress` different from an AlgoTransfer transaction ?

Just ore readable, may not different. For `fundLsig`, can we use args have format like (lsig: string | address,...) depend on type we can choice right actions. In my view, `fundLsig` is enough.What you thinking about it?